src/metabase/driver/generic_sql.clj

```diff
@@ -85,6 +86,9 @@
     "Return a korma form appropriate for converting a Unix timestamp integer field or value to an proper SQL `Timestamp`.
      SECONDS-OR-MILLISECONDS refers to the resolution of the int in question and with be either `:seconds` or `:milliseconds`."))
 
+(extend-protocol jdbc/IResultSetReadColumn
```

Comment:
I'll play around with this a little.  You are right that this will impact all drivers since it applies to the JDBC interface as a whole.